firewood/src/storage/mod.rs

```diff
@@ -723,10 +723,10 @@ impl CachedSpace {
     pub fn new(
         cfg: &StoreConfig,
         disk_requester: DiskBufferRequester,
-    ) -> Result<Self, StoreError<std::io::Error>> {
+    ) -> Result<Arc<Self>, StoreError<std::io::Error>> {
```

Comment:
@rkuris, while not every library does this, I think it's a best practice to follow the standard-library convention of `new` returning `Self` and `try_new` returning `Result<Self, _>`. 

[`Arc<T>` implements `From<T>`](https://doc.rust-lang.org/stable/std/sync/struct.Arc.html#impl-From%3CT%3E-for-Arc%3CT%3E), so if you want to shortcut for `Arc::new`, you can simply call `.into()` on a `CachedSpace` at the call-site. 

@gubatron, could you please change the return type of this function back to what it was? And, if you feel so inclined, feel free to change the method name to `try_new`. 

Thanks again for the contributions!

Reply:
I agree, it was cool but kind of weird.
Love the more "Rustacean" approach of implementing it with the `From<T>` trait.

Reply:
Rust allows us to do this in several ways, I wonder which one you like better:

**Annotating the receiving end and using into**
```rust
let state_cache: Arc<CachedSpace> = CachedSpace::new(
            &StoreConfig::builder()
                .ncached_pages(1)
                .ncached_files(1)
                .space_id(STATE_SPACE)
                .file_nbit(1)
                .rootdir(state_path)
                .build(),
            disk_requester.clone(),
        )
        .unwrap()
        .into();
```

**not annotating the declared variable and using  `.into() as Arc<CachedSpace>`**
```rust
let state_cache = CachedSpace::new(
            &StoreConfig::builder()
                .ncached_pages(1)
                .ncached_files(1)
                .space_id(STATE_SPACE)
                .file_nbit(1)
                .rootdir(state_path)
                .build(),
            disk_requester.clone(),
        )
        .unwrap()
        .into() as Arc<CachedSpace>;
```

**or Using `Arc::from`**
```rust
        let state_cache = Arc::from(
            CachedSpace::new(
                &StoreConfig::builder()
                    .ncached_pages(1)
                    .ncached_files(1)
                    .space_id(STATE_SPACE)
                    .file_nbit(1)
                    .rootdir(state_path)
                    .build(),
                disk_requester.clone(),
            )
            .unwrap(),
        );
```

Reply:
I guess it's not a one fit all solution, using a combination of them